src/coreclr/vm/methodtable.cpp

```diff
@@ -3701,21 +3737,12 @@ int MethodTable::GetRiscV64PassStructInRegisterFlags(CORINFO_CLASS_HANDLE cls)
                     size |= STRUCT_SECOND_FIELD_SIZE_IS8;
                 }
             }
-
-            goto _End_arg;
         }
     }
     else
     {
-        _ASSERTE(th.IsNativeValueType());
+        MethodTable* pMethodTable = th.AsNativeValueType();
 
-        useNativeLayout = true;
-        pMethodTable = th.AsNativeValueType();
-    }
-    _ASSERTE(pMethodTable != nullptr);
```

Comment:
nit: do we remove this one on purpose like: https://github.com/dotnet/runtime/pull/97877/commits/5ec4cb73c894a30a77712f1e96aeb3bae5ebc00f ?